stan/math/prim/cplx/complex.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef STAN_MATH_PRIM_CPLX_COMPLEX_HPP
+#define STAN_MATH_PRIM_CPLX_COMPLEX_HPP
+
+#include <stan/math/prim/cplx/zeroing.hpp>
+#include <complex>
+
+namespace stan {
+namespace math {
+
+/**
+ *  Complex class that forwards the interface of std::complex, brining
+ *  stan::math's namespace into ADL for functions called on data types
+ *  that inherit from this class, as well as allowing template
+ *  specializations of std::complex to indirectly work with different
+ *  underlying data types. For example, std::complex<var> specializes
+ *  the std::complex<T> template to inherit from
+ *  stan::math::complex<T>, which in turn inherits from
+ *  std::complex<zeroing<T>>. This causes stan::math to be pulled into
+ *  ADL on operations involving std::complex<var>, to handle, e.g.
+ *  division operations that the base std::complex doesn't have
+ *  defined usably for var, and, since stan::math::zeroing<var> zero
+ *  initializes itself, will also correctly work with the remaining
+ *  algorithms in std::complex<T> that require the expression T() to
```

Comment:
Clarification? The algorithms expect `T()` to initialize a variable to 0?

Reply:
Yes, that is correct. Some of the algorithms expect T() to initialize to a usable zero.